src/components/AuthForm/AuthForm.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useRef, useState } from "react";
+import { useStore, useDispatch } from "react-redux";
+
+import { fetchLoginData, fetchSignUpData } from "../../ducks/userInfo";
+
+import GoogleSrc from '../../assets/google-symbol.svg';
+
+import {
+    AuthFormContainer,
+    AuthMainContainer,
+    AuthEnterWithGoogle,
+    GoogleIcon,
+    AuthText,
+    InputLabel,
+    ErrorStar,
+    Input,
+    ErrorMsg,
+    AuthActionButtons,
+    Button
+} from "./AuthForm.style";
+
+let formData = {email: null, password: null};
+
+const AuthForm = () => {
+    const store = useStore();
+    const dispatch = useDispatch();
+
+    const emailRef = useRef();
+    const passwordRef = useRef();
+
+    const [validMail, setValidMail] = useState(true);
+    const [validPassword, setValidPassword] = useState(true);
+    const [bluredMail, setBluredMail] = useState(false);
+    const [bluredPassword, setBluredPassword] = useState(false);
+    const [emailErrorMsg, setEmailErrorMsg] = useState("");
+    const [passwordErrorMsg, setPasswordErrorMsg] = useState("");
+
+    const checkEmail = () => {
+        const email = emailRef.current.value;
+
+        if (email.length === 0) {
+            setEmailErrorMsg("это обязательное поле");
+            setValidMail(false);
+        }
+        else if (email.length > 3) {
+            if (!email.match(/[-.\w]+@([\w-]+\.)+[\w-]+/g)) {
+                setEmailErrorMsg("неверный формат почты");
+                setValidMail(false);
+            } else {
+                setValidMail(true);
+                formData.email = email;
+            }
+        }
+        else {
+            setEmailErrorMsg("введите не менее 3 символов");
+            setValidMail(false);
+        }
+    }
+
+    const checkPassword = () => {
+        const password = passwordRef.current.value;
+
+        if (password.length === 0) {
+            setPasswordErrorMsg("это обязательное поле");
+            setValidPassword(false);
+        }
+        else if (password.length > 8) {
+            setValidPassword(true);
+            formData.password = password;
+        }
+        else {
+            setPasswordErrorMsg("введите не менее 8 символов");
+            setValidPassword(false);
+        }
+    }
+
+    const bluredEmailHandler = () => setBluredMail(true);
+
+    const bluredPasswordHandler = () => setBluredPassword(true);
+
+    const checkInputData = (authType) => {
+        if (formData.email && formData.password) {
+            authType === 'login'? dispatch(fetchLoginData(formData)) : dispatch(fetchSignUpData(formData));
+        }
```

Comment:
I think it would be more user friendly if you would show spinner while request is in progress

Reply:
And if user successfully sing in/up it makes sense to redirect him to main page